### ISSUE TITLE
HorizontalPodAutoscaler tries to get resource consumption for Pods that do not exist

### ISSUE THREAD

**Issue by DanielHeckrath** [kubernetes/kubernetes#31584]
**Environment**:
- **Kubernetes version**: 1.3.5
- **Cloud provider or hardware configuration**: Google Container Engine
- **Kernel**: Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2 (2016-04-08) x86_64 GNU/Linux

**What happened**:
We're using a HorizontalPodAutoscaler in our cluster to scale pods based on CPU load. While this is working for some deployments, one deployment seems to be in a broken state where the HPA tries to get cpu consumption from heapster for pods that do not exist.
The HPA was working for said deployment but started to misbehave after a scaling event.

Result from `kubectl get pods`

```
…
gosu-main-api-v1-3735438065-5n8t7                    2/2       Running   1          2h
gosu-main-api-v1-3735438065-j027w                    2/2       Running   1          2h
gosu-main-api-v1-3735438065-rydr6                    2/2       Running   1          2h
gosu-main-api-v1-3735438065-vf59h                    2/2       Running   1          2h
gosu-main-api-v1-3735438065-w4hzm                    2/2       Running   1          2h
…
```

Result from `kubectl get hpa`

```
NAME                      REFERENCE                            TARGET    CURRENT     MINPODS   MAXPODS   AGE
…
gosu-main-api-v1          Deployment/gosu-main-api-v1          75%       <waiting>   3         20        28m
…
```

Result from `kubectl describe hpa gosu-main-api-v1`

```
Name:               gosu-main-api-v1
Namespace:          default
Labels:             <none>
Annotations:            <none>
CreationTimestamp:      Sat, 27 Aug 2016 17:17:31 -0700
Reference:          Deployment/gosu-main-api-v1
Target CPU utilization:     75%
Current CPU utilization:    <unset>
Min replicas:           3
Max replicas:           20
Events:
  FirstSeen LastSeen    Count   From                SubobjectPath   Type        Reason          Message
  --------- --------    -----   ----                -------------   --------    ------          -------
  18m       16m     4   {horizontal-pod-autoscaler }            Warning     FailedGetMetrics    failed to get CPU consumption and request: metrics obtained for 0/12 of pods
  18m       16m     4   {horizontal-pod-autoscaler }            Warning     FailedComputeReplicas   failed to get CPU utilization: failed to get CPU consumption and request: metrics obtained for 0/12 of pods
  30m       30s     58  {horizontal-pod-autoscaler }            Warning     FailedGetMetrics    failed to get CPU consumption and request: metrics obtained for 5/12 of pods
  30m       30s     58  {horizontal-pod-autoscaler }            Warning     FailedComputeReplicas   failed to get CPU utilization: failed to get CPU consumption and request: metrics obtained for 5/12 of pods
```

Excerpt from heapster logs

```
I0828 00:47:34.525902       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-wkcbo
I0828 00:47:34.526057       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-wkcbo
I0828 00:47:34.526107       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-vylz6
I0828 00:47:34.526165       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-vylz6
I0828 00:47:34.526249       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-ipr93
I0828 00:47:34.526304       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-ipr93
I0828 00:47:34.526387       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-qscb9
I0828 00:47:34.526444       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-qscb9
I0828 00:47:34.526493       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-x19pc
I0828 00:47:34.526548       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-x19pc
I0828 00:47:34.526591       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-3b1q3
I0828 00:47:34.526671       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-3b1q3
I0828 00:47:34.526754       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-02y6n
I0828 00:47:34.526822       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-02y6n
I0828 00:48:04.584791       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-x19pc
I0828 00:48:04.584816       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-x19pc
I0828 00:48:04.584837       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-ipr93
I0828 00:48:04.584866       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-ipr93
I0828 00:48:04.584893       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-02y6n
I0828 00:48:04.584961       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-02y6n
I0828 00:48:04.584986       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-wkcbo
I0828 00:48:04.585005       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-wkcbo
I0828 00:48:04.585048       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-1499485845-vylz6
I0828 00:48:04.585066       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-1499485845-vylz6
I0828 00:48:04.585097       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-3b1q3
I0828 00:48:04.585186       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-3b1q3
I0828 00:48:04.585201       1 handlers.go:242] No metrics for container gosu-main-api in pod default/gosu-main-api-v1-786519656-qscb9
I0828 00:48:04.585206       1 handlers.go:178] No metrics for pod default/gosu-main-api-v1-786519656-qscb9
```

**What you expected to happen**:
The deployment is currently set to 5 replicas, so I would expect for the HPA to get the resource consumption for the 5 pods that are scheduled.

**How to reproduce it**:
Cannot reliably bring other deployments into this broken state. Had the same issue with another deployment, but it started to work again without outer influence (maybe after we updated the deployment)

**Comment by DanielHeckrath** [comment]
I was able to fix the problem for my cluster. The pods that the HPA failed to get resource consumption for were stuck in a FAILED state. Those pods were excluded in the result from `kubectl get pods` which lead me to believe that they didn't exist, but I saw them listed in the kube-dashboard by accident.

As a result of this issue, it might be a good idea to either exclude failed pods from resource consumption calculations, or to list them in some form.
